src/tests/account_events.rs

```diff
@@ -0,0 +1,167 @@
+                Some(file.to_str().unwrap()),
+            )
+            .await?;
+        wait_for_item_changed(&context).await;
+        Ok(())
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_set_private_tag() -> Result<()> {
+        let mut tcm = TestContextManager::new();
+        let context = tcm.alice().await;
+        context
+            .set_config(crate::config::Config::PrivateTag, Some("Wonderland"))
+            .await?;
+        wait_for_item_changed(&context).await;
+        Ok(())
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_import_backup() -> Result<()> {
+        let mut tcm = TestContextManager::new();
+        let context1 = tcm.alice().await;
+        let backup_dir = tempfile::tempdir().unwrap();
+        assert!(
+            imex(&context1, ImexMode::ExportBackup, backup_dir.path(), None)
+                .await
+                .is_ok()
+        );
+
+        let context2 = TestContext::new().await;
+        assert!(!context2.is_configured().await?);
+        let backup = has_backup(&context2, backup_dir.path()).await?;
+        imex(&context2, ImexMode::ImportBackup, backup.as_ref(), None).await?;
+        assert!(context2.is_configured().await?);
+        wait_for_item_changed(&context2).await;
+        Ok(())
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_receive_backup() {
+        let mut tcm = TestContextManager::new();
+        // Create first device.
+        let ctx0 = tcm.alice().await;
+        // Prepare to transfer backup.
+        let provider = BackupProvider::prepare(&ctx0).await.unwrap();
+        // Set up second device.
+        let ctx1 = tcm.unconfigured().await;
+
+        ctx1.evtracker.clear_events();
+        get_backup(&ctx1, provider.qr()).await.unwrap();
+
+        // Make sure the provider finishes without an error.
+        tokio::time::timeout(Duration::from_secs(30), provider)
+            .await
+            .expect("timed out")
+            .expect("error in provider");
+
+        wait_for_item_changed(&ctx1).await;
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn test_sync() -> Result<()> {
+        let alice0 = TestContext::new_alice().await;
+        let alice1 = TestContext::new_alice().await;
+        for a in [&alice0, &alice1] {
+            a.set_config_bool(Config::SyncMsgs, true).await?;
+        }
+
+        let new_name = "new name";
+        alice0
+            .set_config(Config::Displayname, Some(new_name))
+            .await?;
+        alice1.evtracker.clear_events();
+        sync(&alice0, &alice1).await;
+        wait_for_item_changed(&alice1).await;
+        assert_eq!(
+            alice1.get_config(Config::Displayname).await?,
+            Some(new_name.to_owned())
+        );
+
+        assert!(alice0.get_config(Config::Selfavatar).await?.is_none());
```

Comment:
I think this may be omitted. There's also `PrivateTag`, but testing events for all synchronised settings looks superfluous, having tests for all settings set locally + one for synchronisation, say, of `Displayname`, is good enough, all the logic for account events is implemented in the only place anyway

Reply:
is private tag synced? I thought it was only local

Reply:
the idea was to test everything that changes in jsonrpc Account type: https://github.com/deltachat/deltachat-core-rust/blob/9812d5ba75482da29ad0b13293ad46120a661dec/deltachat-jsonrpc/src/api/types/account.rs#L11-L26

(except for address and color, address is not shown anymore in account list and I doubt that color changes that often (maybe with aeap when address changes? but we can add a test for that then, when this is more common))

Reply:
Ok, private tag isn't synced, but this isn't important. The idea is to avoid combinatorial explosion of testcases:
- We already have tests on synchronisation of `Displayname` and `Selfavatar`
- You add also tests that account events for them are emitted as expected
- Then it's sufficient to have only one test that account events are emitted on the second device for a synchronised setting, no need to test this for all settings, moreover all the core logic is implemented in the only place

Reply:
I don't agree that tests should rely on implementation details or on other tests of different files. IMO tests should test all cases. how it's implemented shouldn't matter to the tests.

We could add the event check to the old test and remove the new test, but then we have less separation of concern - so I think it's ok to have duplicated test code.

Reply:
I also don't like an idea of "only white-box testing", i just can't imagine how this code block can test anything in addition to other tests (i.e. how the core code should be changed to create a new uncovered path). And if we don't want to rely on what other tests do, we always need to go through all cases combinatorially which is just impossible.

> We could add the event check to the old test and remove the new test

I don't suggest to remove the whole test, only the block for `Selfavatar`. And yes, we can add event checks to some existing tests (even just in case), that will still be less code.

This is not critical for merging this PR of course, i just suggest to find some practical compromise for the amount of test code.

Reply:
we could remove the checks if it worked and just test if the action triggered the event. But I still think the more tests the better.
Also this is not very hard to maintain code in my view, the code is straight forward and easy to understand in this case.

Reply:
i am lost a bit in the exact discussion, but, in general, i agree with @Simon-Laux - the more tests the better. 

when i write tests  for sth, i do not care if the things i assert() are already checked elsewhere, or if some assert()s are not strictly related. i even think, there is worth in different tests about the same thing, some test may change - of do not cover the exact same cornercase, this is not always obvious, and often also not worth to think about.

i would also not like to have too much conventions/compliance how to do tests etc. -- that would only lead to less tests written. review should focus on the implementation mostly - and complain about _missing_ tests